### PR TITLE
Tests for queue empty oneAPI extension

### PR DESCRIPTION
**Pull request by vladimirkhashev**
Added tests for queue empty oneapi extension according to the [test plan](https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-2020/test_plans/oneapi_queue_empty.asciidoc).

### CODE REVIEW

File: tests/extension/oneapi_queue_empty/queue_empty_ext_test.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+TEST_CASE(
+    "Test for \"Queue empty\" extension, check extension functionality"
+    "after commands are submitted to the queue, test 1",
+    "[oneapi_queue_empty]") {
+#ifndef SYCL_EXT_ONEAPI_QUEUE_EMPTY
+  SKIP("SYCL_EXT_ONEAPI_QUEUE_EMPTY is not defined");
+#else
+  auto queue = sycl_cts::util::get_cts_object::queue();
+
+  size_t array[buf_size];
```
Comment (keryell) [inline tests/extension/oneapi_queue_empty/queue_empty_ext_test.cpp:63]: Do you need these arrays?

File: tests/extension/oneapi_queue_empty/queue_empty_ext_test.cpp [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  sycl::buffer<size_t, 1> array_buf(array, buf_size);
+
+  auto e = queue.submit([&](sycl::handler& cgh) {
+    auto array_acc =
+        array_buf.template get_access<sycl::access_mode::write>(cgh);
```
Comment (keryell) [inline tests/extension/oneapi_queue_empty/queue_empty_ext_test.cpp:68]: Use `sycl::accessor` constructor instead everywhere.

**Review by keryell — CHANGES_REQUESTED** [review]

File: tests/extension/oneapi_queue_empty/queue_empty_ext_test.cpp (hunk 2 above)
Comment (vladimirkhashev) [reply tests/extension/oneapi_queue_empty/queue_empty_ext_test.cpp]: Done.

File: tests/extension/oneapi_queue_empty/queue_empty_ext_test.cpp (hunk 1 above)
Comment (vladimirkhashev) [reply tests/extension/oneapi_queue_empty/queue_empty_ext_test.cpp:63]: I added these arrays to avoid compiler optimizations in kernel code.

**Comment by bader** [comment]
@keryell, ping.

File: tests/extension/oneapi_queue_empty/queue_empty_ext_test.cpp (hunk 1 above)
Comment (keryell) [reply tests/extension/oneapi_queue_empty/queue_empty_ext_test.cpp:63]: Which compiler optimization?
But my point is that would it work with just
```c++
sycl::buffer<size_t> array_buf(buf_size);
```
instead?

File: tests/extension/oneapi_queue_empty/queue_empty_ext_test.cpp [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    "Test for \"Queue empty\" extension, check extension functionality"
+    "after commands are submitted to the queue, test 1",
+    "[oneapi_queue_empty]") {
+#ifndef SYCL_EXT_ONEAPI_QUEUE_EMPTY
+  SKIP("SYCL_EXT_ONEAPI_QUEUE_EMPTY is not defined");
+#else
+  auto queue = sycl_cts::util::get_cts_object::queue();
+
+  size_t array[buf_size];
+  sycl::buffer<size_t, 1> array_buf(array, buf_size);
```
Comment (bader) [inline tests/extension/oneapi_queue_empty/queue_empty_ext_test.cpp:64]:
```suggestion
  sycl::buffer<size_t, 1> array_buf(buf_size);
```

File: tests/extension/oneapi_queue_empty/queue_empty_ext_test.cpp (hunk 1 above)
Comment (bader) [reply tests/extension/oneapi_queue_empty/queue_empty_ext_test.cpp]: Done.

**Review by keryell — APPROVED** [review]
Thanks.

**Pull request merged**